pkg/api/controller.go

```diff
@@ -2648,6 +2650,10 @@ func (c *Controller) UploadObject(w http.ResponseWriter, r *http.Request, reposi
 	} else {
 		entryBuilder.AddressType(catalog.AddressTypeFull)
 	}
+	meta := extractLakeFSMetadata(r.Header)
+	if meta != nil {
```

Comment:
Use `len` it will catch nil and empty map (which is also the case for `extractLakeFSMetadata`